ci: automate PR for sdk version bump

efgpinto opened the pull request:
As draft for now to be able to test on CI

efgpinto commented on .circleci/config.yml:
```diff
@@ -209,6 +210,31 @@ jobs:
+            BRANCH=bump-sdk-versions-$SDK_VERSION
+            git checkout -b $BRANCH
+            git config user.name "Kalix Bot"
+            git config user.email "kalix.team@lightbend.com" #FIXME use a proper email
```
Waiting to check what email we want to use here

ennru commented:
This may affect https://github.com/lightbend/kalix-jvm-sdk/pull/1021

efgpinto commented:
> This may affect #1021

Yeah, somewhat related but shouldn't conflict. I've decided to a different script because the existing one does more than needed (installing the version locally).. but we could also opt to make this new one parameterizable and make the other one depend on this. I deemed it may not be worth the effort but open for discussion.

efgpinto commented on .circleci/config.yml:
```diff
@@ -209,6 +210,31 @@ jobs:
+      - run:
+          name: Create PR to update all sdk versions (samples, sbt and maven)
+          command: |
+            chmod +x ./updateSdkReleasedVersion.sh
```
not sure why it required this, it should be runable without it I believe.. but CI complained about it when I tested.

efgpinto commented on updateSdkReleasedVersion.sh:
```diff
@@ -0,0 +1,20 @@
+find ./samples -type f -name "*plugins.sbt" -print0 | xargs -0 sed -i -e "s/System.getProperty(\"kalix-sdk.version\", \".*\"))/System.getProperty(\"kalix-sdk.version\", \"$SDK_VERSION\"))/"
+
+echo ">>> Updating maven plugin to $SDK_VERSION"
+cd maven-java && mvn versions:set -DnewVersion="$SDK_VERSION"
```
I guess at some point we could probably consider having a `scripts` folder instead..

franciscolopezsancho replied:
I think this makes sense. In `kalix-proxy` we have one `bin` folder for that.

franciscolopezsancho commented:
Combining both scripts into one I also think is a good idea. Maybe it is better to do it now than put it off and perhaps forget it.

franciscolopezsancho commented:
If you decide to merge both I'll wait and test it afterwards.

efgpinto commented:
Updated now, added some usage examples to PR description for clarity.

Also, made the `updatePomVersions.sh` depend on this one... TBH, I think that one should probably go but I was losing a bit too much time to think about the params we would need on this new one to support installing the local version vs just updating it.. So I'm not sure we need to do that here since we also have https://github.com/lightbend/kalix-jvm-sdk/pull/1021 open.

Unless someone has already a clear idea of what the this command could look like to support both features, I'm tempted to go with this version so we have the auto-PR thing in place and then try to re-org these 3 scripts (`updateSdkVersions`, `updatePomVersions` and `publishLocalM2`).

Thoughts?

efgpinto commented:
@johanandren IIUC you're making an sdk release soon. If you want to have a look at this one prior to that, hopefully it can make the release a bit easier.

johanandren commented:
@efgpinto failed because of bot not having permissions:

https://app.circleci.com/pipelines/github/lightbend/kalix-jvm-sdk/4386/workflows/a6b43db1-3885-4519-add4-0dc9fec1c74d/jobs/67064

efgpinto commented:
> @efgpinto failed because of bot not having permissions:
> 
> https://app.circleci.com/pipelines/github/lightbend/kalix-jvm-sdk/4386/workflows/a6b43db1-3885-4519-add4-0dc9fec1c74d/jobs/67064

Thanks for reviewing and testing Johan. Seems like the bot's permissions got downgraded, re-run solved that but the created [PR](https://github.com/lightbend/kalix-jvm-sdk/pull/1177) was not perfect. Will have a look.
